types/index.d.ts

```diff
@@ -0,0 +1,151 @@
+}
+
+export interface Encoder {
+	// TODO
+}
+
+export interface ReaderOptions {
+	// TODO
+}
+
+interface AssembleOptions {
+	importHook: (filePath: string, type: 'idl', callback: Callback<object>) => void;
+}
+
+export function assembleProtocol(filePath: string, opts: AssembleOptions, callback: Callback<object>): void;
+export function assembleProtocol(filePath: string, callback: Callback<object>): void;
+export function createFileDecoder(fileName: string, codecs?: CodecOptions): Decoder;
+export function createFileEncoder(filePath: string, schema: any, options?: any): Encoder;
+export function discoverProtocol(transport: Service.Transport, options: any, callback: Callback<any>): void;
+export function discoverProtocol(transport: Service.Transport, callback: Callback<any>): void;
+export function extractFileHeader(filePath: string, options?: any): void;
+export function parse(schemaOrProtocolIdl: string, options?: any): Service.Protocol | Type.Type; // TODO
+export function readProtocol(protocolIdl: string, options?: ReaderOptions): Service.Protocol;
+export function readSchema(schemaIdl: string, options?: ReaderOptions): Type.Schema;
+// TODO streams
+// TODO types
+
+export namespace Type {
+	type Schema = string;
+
+	interface Type {
+		// TODO clone(val, opts)
+		// TODO compare
+		// TODO compareBuffers(buf1, buf2)
+		// TODO createResolver(type, opts)
+		// TODO decode(buf, pos, resolver)
+		// TODO encode(val, buf, pos)
+		// TODO equals(type)
+		// TODO fingerprint(algorithm)
+		fromBuffer(buffer: Buffer, resolver: any, noCheck: boolean): Type; // TODO
+		// TODO fromString(str)
+		// TODO inspect()
+		// TODO isValid(val, opts)
+		// TODO random()
+		// TODO schema(opts)
+		toBuffer(value: object): Buffer;
+		// TODO toJSON()
+		// TODO toString(val)
+		// TODO wrap(val)
+	}
+
+	function forSchema(schema: Schema): Type;
+	function forValue(value: object): Type;
+	// TODO function forTypes(types, opts)
+	function isType(arg: any): boolean; // TODO
+}
+
+export namespace Service {
+	interface Protocol {
+	}
+
+	interface ClientChannel extends EventEmitter {
+		readonly client: Client;
+		readonly timeout: number;
+		readonly destroyed: boolean;
+		readonly draining: boolean;
+		readonly pending: number;
+		ping(timeout: number, cb: any): void;
+		destroy(noWait: boolean): void;
+	}
+
+	interface ServerChannel extends EventEmitter  {
+		readonly server: Server;
+		readonly destroyed: boolean;
+		readonly draining: boolean;
+		readonly pending: number;
+		destroy(noWait: boolean): void;
+	}
+
+	interface ClientOptions {
+		buffering: boolean;
+		cache: any;
+		channelPolicy: any;
+		strictTypes: boolean;
+		timeout: number;
+		remoteProtocols: boolean;
+	}
+
+	interface ServerOptions {
+		objectMode: boolean;
+	}
+
+	interface Service {
+		createClient(options?: ClientOptions): Client;
+		createServer(options?: ServerOptions): Server;
+		message(name: string): any;
+		type(name: string): any;
+		inspect(name: string): any;
```

Comment:
I'm not sure it's useful to include this method, it's an override just for pretty-printing in the REPL.